op-e2e/e2eutils/io.go

```diff
@@ -0,0 +1,27 @@
+package e2eutils
+
+import (
+	"crypto/rand"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TempDir(t *testing.T) string {
+	suffix := make([]byte, 8)
+	_, err := rand.Read(suffix)
+	require.NoError(t, err)
+
+	dir := filepath.Join(os.TempDir(), fmt.Sprintf("op-e2e-%s-%x", t.Name(), suffix))
+	require.NoError(t, os.MkdirAll(dir, 0755))
+	t.Cleanup(func() {
+		err := os.RemoveAll(dir)
+		if err != nil {
+			t.Logf("Error removing temp dir %s: %s", dir, err)
+		}
+	})
+
+	return dir
+}
```

Comment:
I'm a bit nervous about this. It doesn't have the same guarantee of uniqueness that creating a temp file does. It should be fine given the test name is in there but it's an unfortunate direction to head down.

The big question for me is why geth is still writing to the data dir when its process should have already been terminated. I'm a bit torn on whether we merge this to stop the intermittency or if that's just papering over another issue somewhere.

Reply:
Looking at the implementation of `os.MkdirTemp()`, (which the `t.TempDir()` method uses under the hood), it adds a random suffix and iterates until it finds one that doesn't exist. I've updated this to use `MkdirTemp` which will give us the same guarantees.

Re: why it's writing to the datadir - it seems to be solely related to the external geth instance tests. We currently swallow errors when we attempt to close the external Geth instance. I've added some additional logging so we can capture why it's not shutting down. The shutdown algorithm right now is:

1. Send a sigterm
2. If no sigterm is received in 5s, then send a sigkill after 30 seconds
3. If the process still isn't dead, return an error

I don't see and `caught signal` messages in the CircleCI logs, so it's possible the signal is being dropped somewhere. Right now there's not a lot to go on other than that, though, which is why I added the logging.